tests/unit_tests/test_media/test_media_logging.py

```diff
@@ -0,0 +1,142 @@
+    return wandb.Image(np.ones(shape=(32, 32)))
+
+
+@pytest.fixture
+def table_media() -> wandb.Table:
+    return wandb.Table(data=[[1]], columns=["A"])
+
+
+@pytest.fixture
+def graph_media() -> wandb.Graph:
+    graph = wandb.Graph()
+    node_a = data_types.Node("a", "Node A", size=(4,))
+    node_b = data_types.Node("b", "Node B", size=(16,))
+    graph.add_node(node_a)
+    graph.add_node(node_b)
+    graph.add_edge(node_a, node_b)
+    return graph
+
+
+@pytest.fixture
+def bokeh_media() -> data_types.Bokeh:
+    x = [1, 2]
+    y = [6, 7]
+    p = figure(title="simple line example", x_axis_label="x", y_axis_label="y")
+    p.line(x, y, legend_label="Temp.", line_width=2)
+    return data_types.Bokeh(p)
+
+
+@pytest.fixture
+def html_media() -> wandb.Html:
+    return wandb.Html("<html><body><h1>Hello, World!</h1></body></html>")
+
+
+@pytest.fixture
+def molecule_media() -> wandb.Molecule:
+    m = Chem.MolFromSmiles("Cc1ccccc1")
+    return wandb.Molecule.from_rdkit(m)
+
+
+@pytest.fixture
+def object3d_media() -> wandb.Object3D:
+    point_cloud = np.random.rand(100, 3)
+    return wandb.Object3D(point_cloud)
+
+
+@pytest.fixture
+def plotly_media() -> wandb.Plotly:
+    fig, ax = plt.subplots(2)
+    ax[0].plot([1, 2, 3])
+    ax[1].plot([1, 2, 3])
+    wandb.Plotly.make_plot_media(plt)
+    return wandb.Plotly(fig)
+
+
+@pytest.mark.parametrize(
+    "media_object",
+    [
+        "table_media",
+        "image_media",
+        "video_media",
+        "audio_media",
+        "graph_media",
+        "bokeh_media",
+        "html_media",
+        "molecule_media",
+        "object3d_media",
+        "plotly_media",
+    ],
+)
+def test_log_media_saves_to_run_directory(mock_run, request, media_object):
+    run = mock_run(use_magic_mock=True)
+    media_object = request.getfixturevalue(media_object)
+
+    media_object.bind_to_run(run, "/media/path", 0)
+
+    # Assert media object is saved under the run directory
+    assert media_object._path.startswith(run.dir)
+
+
+@pytest.mark.parametrize(
+    "invalid_character",
+    [
+        "<",
+        ">",
+        ":",
+        "\\",
+        "?",
+        "*",
+    ],
+)
+@pytest.mark.skipif(platform.system() != "Windows", reason="Windows-specific test")
+def test_log_media_with_invalid_character_on_windows(
+    mock_run, image_media, invalid_character
+):
+    run = mock_run()
+    with pytest.raises(ValueError, match="Media .* is invalid"):
+        image_media.bind_to_run(run, f"image{invalid_character}test", 0)
+
+
+def test_log_media_with_path_traversal(mock_run, image_media):
+    run = mock_run()
+    image_media.bind_to_run(run, "../../../image", 0)
+
+    # Resolve to path to verify no path traversals
+    resolved_path = str(Path(image_media._path).resolve())
+    assert resolved_path.startswith(run.dir)
+
+
+def test_log_media_prefixed_with_multiple_slashes(mock_run, image_media):
+    run = mock_run()
+    image_media.bind_to_run(run, "////image", 0)
```

Comment:
Can you check `my////image` as well? I'm not sure that case is handled.

Reply:
Ahh I did test this manually but missed that exact case in the tests. Added a test for the case you provided.

This case doesn't need to be explicitly handled. This is due to the fact that the os appears to ignore multiple `/` when in a path.

Reply:
That makes sense! I think a test is good to have just to document that we're aware of and rely on that behavior. I only point this out because that's a more nuanced aspect of file paths which isn't obviously handled the same way by all OSes (it seems to be, but I had to look it up).